File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <deque>
using namespace std;
deque<int> dq;
int solution(vector<int> people, int limit)
{
    int answer = 0;
    copy(people.begin(), people.end(), inserter(dq, dq.end()));

    sort(dq.begin(), dq.end());

    while (dq.size() != 0)
    {
        if (dq.size() == 1)
        {
            answer++;
            break;
        }
        if (dq[0] + dq[dq.size() - 1] <= limit)
        {
            dq.pop_front();
            dq.pop_back();
            answer++;
        }
        else
        {
            dq.pop_back();
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;
class Solution {

    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int front = 0;
        int rear = people.length - 1;

        int count = 0;
        while (front <= rear) {
            if (people[front] + people[rear] <= limit) {
                front++;
                rear--;
            } else {
                rear--;
            }
            ++count;
        }

        return count;
    }
}